virtual_ecosystem/models/abiotic/abiotic_model.py

```diff
@@ -197,15 +183,26 @@ def setup(self) -> None:
         TODO: Remove when the base model is updated.
         """
 
-    def _setup(self) -> None:
+    def _setup(
+        self, model_constants: AbioticConsts = AbioticConsts(), **kwargs
+    ) -> None:
         """Function to set up the abiotic model.
 
         This function initializes soil temperature and canopy temperature for all
         corresponding layers and calculates the reference vapour pressure deficit for
         all time steps of the simulation. All variables are added directly to the
         self.data object.
+
+        Args:
+            model_constants: Set of constants for the abiotic model.
+            **kwargs: Further arguments to the setup method.
         """
 
+        self.model_constants: AbioticConsts = model_constants
+        """Set of constants for the abiotic model."""
```

Comment:
Presumably this will never be changed after assignment? Just checking because we don't want the default arg above to be reused otherwise

Reply:
All the constant classes are frozen dataclasses, so they cannot be changed.